clinica/utils/io.py

```diff
@@ -189,12 +192,25 @@ def check_input_caps_files(list_caps_files, caps_type, pipeline_name, caps_direc
                 "\n* %s%s%s. Found files:\n%s" % (
                     Fore.BLUE, id_duplicated_file.replace('_', '|'), Fore.RESET,
                     '\n'.join('- ' + s for s in list_caps_files if id_duplicated_file in s)
-                    # (Fore.BLUE + '|' + Fore.RESET).join(s for s in list_caps_files if id_duplicated_file in s)
                     ))
         error_message += (
             "\n%s* Error explanations:\n"
             " - Clinica expected to find a single file in CAPS directory for each subject.\n"
             " - Did you duplicate files in CAPS directory?%s\n" %
             (Fore.YELLOW, Fore.RESET)
         )
+    if not error_message:
+        if not (subject_ids == id_caps_files):
+            from clinica.utils.stream import cprint
+            cprint(subject_ids)
+            cprint(id_caps_files)
+            error_message += (
+                    "\n%s[Error] Order between subjects given for Clinica and %s CAPS files mismatch:%s\n" %
```

Comment:
Rather than telling the user that there is a mismatch, you should rearrange the list of files otherwise the user is stuck with an error he has no power to deal with.

Reply:
I kind of disagree on this one: if this situation occurs, this means that something went wrong either with the tool/function that grabs BIDS/CAPS files or there is an issue in the BIDS/CAPS dataset.

In the first case, we would have to fix the tool/function. In the second case, I think we need to warn the user about it.

Reply:
After discussion with @arnaudmarcoux, it turns out that I didn't pay attention that when using PyBIDS or PyCAPS, it grabbed more files than expected due to permissive regexp behind.

As advised, 474538a now reorders BIDS or CAPS files based on input IDs at the beginning of the pipeline.